src/clients/c/samples/main.c

```diff
@@ -0,0 +1,435 @@
+    );
+
+    if (status != TB_STATUS_SUCCESS) {
+        printf("Failed to initialize tb_client\n");
+        exit(-1);
+    }
+
+    completion_context_t ctx;
+    completion_context_init(&ctx);
+
+    tb_packet_t *packet;
+    tb_packet_list_t packet_list;
+
+    ////////////////////////////////////////////////////////////
+    // Submitting a batch of accounts:                        //
+    ////////////////////////////////////////////////////////////
+
+    #define ACCOUNTS_LEN 2
+    #define ACCOUNTS_SIZE sizeof(tb_account_t) * ACCOUNTS_LEN
+    tb_account_t accounts[ACCOUNTS_LEN];
+    
+    // Zeroing the memory, so we don't have to initialize every field.
+    memset(&accounts, 0, ACCOUNTS_SIZE);
+    
+    accounts[0].id = 1;
+    accounts[0].code = 2;
+    accounts[0].ledger = 777;
+
+    accounts[1].id = 2;
+    accounts[1].code = 2;
+    accounts[1].ledger = 777;
+    
+    // Acquiring a packet for this request:
+    packet = acquire_packet(&packets_pool);
+    packet->operation = TB_OPERATION_CREATE_ACCOUNTS;  // The operation to be performed.
+    packet->data = accounts;                           // The data to be sent.
+    packet->data_size = ACCOUNTS_SIZE;                 //
+    packet->user_data = &ctx;                          // User-defined context.
+    packet->status = TB_PACKET_OK;                     // Will be set when the reply arrives.
+
+    printf("Creating accounts...\n"); 
+    
+    packet_list.head = packet;
+    packet_list.tail = packet;
+    send_request(client, &packet_list, &ctx);
+
+    if (packet->status != TB_PACKET_OK) {
+        // Checking if the request failed:
+        printf("Error calling create_accounts (ret=%d)\n", packet->status);
+        exit(-1);
+    }
+
+    if (ctx.size != 0) {
+        // Checking for errors creating the accounts:
+        tb_create_accounts_result_t *results = (tb_create_accounts_result_t*)ctx.reply;
+        int results_len = ctx.size / sizeof(tb_create_accounts_result_t);
+        printf("create_account results:\n");
+        for(int i=0;i<results_len;i++) {
+            printf("index=%d, ret=%d\n", results[i].index, results[i].result);
+        }
+        exit(-1);
+    }
+
+    // Releasing the packet, so it can be used in a next request.
+    release_packet(&packets_pool, packet);
+
+    printf("Accounts created successfully\n");
+    
+    ////////////////////////////////////////////////////////////
+    // Submitting multiple batches of transfers:              //
+    ////////////////////////////////////////////////////////////
+
+    printf("Creating transfers...\n");
+    #define MAX_BATCHES 100
+    #define TRANSFERS_PER_BATCH ((MAX_MESSAGE_SIZE) / sizeof(tb_transfer_t))
+    long max_latency_ms = 0;
+    long total_time_ms = 0;
+    for (int i=0; i< MAX_BATCHES;i++) {
+        tb_transfer_t transfers[TRANSFERS_PER_BATCH];
+        // Zeroing the memory, so we don't have to initialize every field.
+        memset(transfers, 0, MAX_MESSAGE_SIZE);
```

Comment:
Potential OOB write if `MAX_MESSAGE_SIZE` is no longer exactly divisible by `sizeof(tb_transfer_t)`. Might want to make a `#define MAX_TRANSFER_SIZE (sizeof(tb_transfer_t) * TRANSFERS_PER_BATCH)`.